### PR TITLE
[v7r3] Python 3 server installation docs

### PR DESCRIPTION
**Pull request by chrisburr**
Based on #5171 with two additional commits to add the Python 3 server installation instructions. The first commit mostly duplicates the Python 2 instructions so the actual differences can be seen in 3d00336.

### CODE REVIEW

**Comment by fstagni** [comment]
Since I am the author of many of the commits in this PR, I can't really review. 
@andresailer and/or @atsareg and/or @marianne013 please leave your review.

**Comment by chrisburr** [comment]
I'm converting this to a draft as I think the Python 3 server install docs might change slightly and I'd rather test them before merging so the corrections don't get forgotten.

**Comment by andresailer** [comment]
I pushed a bit of syntax fixes for the Webapp UserGuide

File: docs/source/UserGuide/WebAppUserGuide/index.rst [hunk 1]
```diff
@@ -38,47 +31,317 @@ Two protocols are allowed: **http** and **https**.
 The state of applications or desktops can not be saved.
 **https** protocol allows to access all functionalities of DIRAC depending on your role (DIRAC group). 
 The state of the application is not saved in the **URL**. The URL only contains the name of application or desktop. 
-For example: `https://lhcb-portal-dirac.cern.ch/DIRAC/s:LHCb-Production/g:lhcb_prmgr/?view=tabs&theme=Grey&url_state=1|AllPlots`   
+For example: ``https://lhcb-portal-dirac.cern.ch/DIRAC/s:LHCb-Production/g:lhcb_prmgr/?view=tabs&theme=Grey&url_state=1|AllPlots``
```
Comment (fstagni) [inline docs/source/UserGuide/WebAppUserGuide/index.rst:34]: I think it becomes

```suggestion
For example: ``https://lhcb-portal-dirac.cern.ch/DIRAC/s:LHCb-Production/g:lhcb_prmgr/?theme=Grey&url_state=1|AllPlots``
```

File: docs/source/UserGuide/WebAppUserGuide/index.rst [hunk 2]
```diff
@@ -38,47 +31,317 @@ Two protocols are allowed: **http** and **https**.
...
-      #. The state is an application: *LHCbDIRAC.LHCbJobMonitor.classes.LHCbJobMonitor:AllUserJobs,*
-   
+
+Format of the URL:
+
+#. https://: protocol
+#. lhcb-portal-dirac.cern.ch/DIRAC/: host.
+#. s:LHCb-Production: DIRAC setup.
+#. g:lhcb_prmgr : role
+#. view=tabs : it is the theme.
```
Comment (fstagni) [inline docs/source/UserGuide/WebAppUserGuide/index.rst:44]: remove this line.

File: docs/source/UserGuide/WebAppUserGuide/index.rst [hunk 3]
```diff
@@ -38,47 +31,317 @@ Two protocols are allowed: **http** and **https**.
...
+* DIRAC: is an external link to DIRAC portal
+* My Desktops: contains all saved desktops. You can see a **Default** desktop which contains
+  all applications which belongs to the **Default** desktop.
+* Shared: It contains all Shared desktops and applications.
+
+
+Manage application and desktop
+------------------------------
+
+You can manage the state of applications and desktops by by clicking to the following menu.
```
Comment (fstagni) [inline docs/source/UserGuide/WebAppUserGuide/index.rst:122]:
```suggestion
You can manage the state of applications and desktops by clicking to the following menu.
```

File: src/DIRAC/Core/scripts/install_full_py3.cfg [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+  #
+  #  Name of the Admin user (default: None )
+  AdminUserName = zmathe
+  #  DN of the Admin user certificate (default: None )
+  #  In order the find out the DN that needs to be included in the Configuration for a given
+  #  host or user certificate the following command can be used::
+  #
+  #          openssl x509 -noout -subject -enddate -in <certfile.pem>
+  #
+  AdminUserDN = /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=zmathe/CN=674937/CN=Zoltan Mathe
```
Comment (fstagni) [inline src/DIRAC/Core/scripts/install_full_py3.cfg:50]: put John Doe?

File: src/DIRAC/Core/scripts/install_full_py3.cfg [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+  ConfigurationName = Dirac-Production
+  #
+  #   These options define the DIRAC components to be installed on "this" DIRAC server.
+  #
+  #
+  #  The next options should only be set for the primary server,
+  #  they properly initialize the configuration data
+  #
+  #  Name of the Admin user (default: None )
+  AdminUserName = zmathe
```
Comment (chrisburr) [inline src/DIRAC/Core/scripts/install_full_py3.cfg:43]:
```suggestion
  AdminUserName = jdoe
```

File: src/DIRAC/Core/scripts/install_full_py3.cfg [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+  #  DN of the Admin user certificate (default: None )
+  #  In order the find out the DN that needs to be included in the Configuration for a given
+  #  host or user certificate the following command can be used::
+  #
+  #          openssl x509 -noout -subject -enddate -in <certfile.pem>
+  #
+  AdminUserDN = /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=zmathe/CN=674937/CN=Zoltan Mathe
+  #  Email of the Admin user (default: None )
+  AdminUserEmail = zoltan.mathe@cern.ch
```
Comment (chrisburr) [inline src/DIRAC/Core/scripts/install_full_py3.cfg:52]:
```suggestion
  AdminUserEmail = john.doe@example.invalid
```

File: docs/source/UserGuide/WebAppUserGuide/index.rst [hunk 7]
```diff
@@ -38,47 +31,317 @@ Two protocols are allowed: **http** and **https**.
...
+you want to share (more information above in the `Manage application and desktop`_).
+
+**Share an application/desktop**
+
+You have to do the following steps to share an application/desktop:
+
+#. right click on the desktop/application what you want to share.
+#. choose the menu item: Share desktop or Share Application.
+#. copy the text (for example: desktop|zmathe|lhcb_prmgr|JobMonitorAll) and click OK on the pop up window:
+#. send the text (desktop|zmathe|lhcb_prmgr|JobMonitorAll) to the person
```
Comment (chrisburr) [inline docs/source/UserGuide/WebAppUserGuide/index.rst:224]:
```suggestion
#. copy the text (for example: desktop|jdoe|lhcb_prmgr|JobMonitorAll) and click OK on the pop up window:
#. send the text (desktop|jdoe|lhcb_prmgr|JobMonitorAll) to the person
```

File: docs/source/UserGuide/WebAppUserGuide/index.rst [hunk 8]
```diff
@@ -38,47 +31,317 @@ Two protocols are allowed: **http** and **https**.
...
+   :align: center
+
+The State Loader widget is the following:
+
+.. image:: images/loader.png
+   :scale: 50 %
+   :alt: Loader.
+   :align: center
+
+You have to provide the Shared State (for example: desktop|zmathe|lhcb_prmgr|JobMonitorAll) and a name (for example: newName).
```
Comment (chrisburr) [inline docs/source/UserGuide/WebAppUserGuide/index.rst:247]:
```suggestion
You have to provide the Shared State (for example: desktop|jdoe|lhcb_prmgr|JobMonitorAll) and a name (for example: newName).
```

**Comment by chrisburr** [comment]
Assuming I didn't break anything in the last commit this PR is ready. The content is mostly from @fstagni's earlier PR in #5171. The Python 3 specific stuff I added can be found by looking at the commit history.

**Review by fstagni — APPROVED** [review]

**Pull request merged**